### ISSUE TITLE
Slightly unreasonable default alignment for multiple windows which do not fill the screen.

### ISSUE THREAD

**Issue by maralorn** [dawsers/hyprscroller#80]
Hey there. Thanks for writing hyprscroller! And I am totally fine with you not taking feature requests.

However I have a big monitor and with two columns which don’t fill the screen I observe this:

```
| focused w1 | unfocused w2 | ... empty space .... |
```
When I then focus w2 I get:

```
| ... empty space ... | unfocused w1 | focused w2|
```
so with multiple windows which don’t fill the screen, they unnecessarily bounce around left and right always moving the focused window out of the middle of the screen. That’s a bit weird. :laughing:

I might look into fixing this myself, at some point.

**Comment by dawsers** [comment]
That's a bit strange, focus shouldn't move windows if all fit the viewport. What version of Hyprland are you using?

For example:
- I create one window. It is `onehalf` (my default) and centered on the viewport.
- I `cyclesize` it `prev` so it is `onethird`.
- I create another window and repeat the sizing.
- Now I have two windows centered on the screen with some space on the beginning and end of the monitor (one sixth on each side).
- I can change focus from one to the other and nothing moves. Focus is semi-lazy, if everything fits the screen, nothing moves.

Please, tell me your version and test what I explained above.

**Comment by maralorn** [comment]
Ah, good to know, that you are not seeing this.

I have tried your description. The first window is centered in the screen. In the moment I create the second window (I have onefourth as default) it pops up at the right of the first one and both get pulled to the right border. Switching focus to the left makes the windows bounce from the right to the left (and this can be repeated) although both windows are in full view all the time.

I am on nixos 24.11, which has hyprland 0.45.0 and commit 556e01458ba26aa63253ae590a6aa8b2601ecf03 from Nov 9th of hyprscroller.

**Comment by dawsers** [comment]
Maybe there was a bug at that time, I made some changes to window recalculation.

If I were you, I would either compile manually and try the latest version, which should still be compatible with 45.0, or even better, upgrade to Hyprland 0.45.2. There are two changes I made there that will make hyprscroller work better.

**Comment by MattKang** [comment]
Was just coming here to investigate the same thing. Also on nix 24.11 and hyprland 0.45.0. Thanks!

**Comment by dawsers** [comment]
Strictly speaking, I only do bug support for the latest stable Hyprland version, no git and no older versions. That said, most stable Hyprland versions have a version of *hyprscroller* that is quite stable, as there is usually around a month between releases, enough time to squash the most obvious bugs.

As I expected most distros to update Hyprland on point releases (v0.45.0 to v0.45.1 to v0..45.2), I didn't bother transporting the commit pin to those. There are clear reasons why it is good to be at the latest point release of your Hyprland version. If the maintainer decides a point release is needed, it is usually because there was an important change that was worth backporting, because doing it is not fun. I tried to favor that behavior by not updating the pins.

But I see Nix does its own thing. It is trivial for me to simply copy the commit pin to v0.45.0 (compatible API), and seeing this, I will do it. But I think it would be better for Nix to update to those point releases. If vaxry took the trouble to release them, there must be a good reason.

Another matter is Hyprland v0.45.0 has a bug that may cause *hyprscroller* to crash when in scaled overview mode if your windows become too small. I submitted a PR and it was backported to v0.45.2, so there is an additional reason to upgrade.

But I also don't want to cause any trouble to anybody who is stuck in v0.45.0, as long as they know the issues above, so...done in 447e001be3ea207fdbd7e9281df6ad5a91fe6b3a
